x-pack/plugins/enterprise_search/server/routes/app_search/search.ts

```diff
@@ -36,4 +38,25 @@ export function registerSearchRoutes({
       path: '/api/as/v1/engines/:engineName/search.json',
     })
   );
+
+  // For the Search UI routes below, Search UI always uses the full API path, like:
+  // "/api/as/v1/engines/{engineName}/search.json". We only have control over the base path
+  // in Search UI, so we created a common basepath of "/api/app_search/search-ui" here that
+  // Search UI can use.
+  //
+  // Search UI *also* uses the click tracking and query suggestion endpoints, however, since the
+  // App Search plugin doesn't use that portion of Search UI, we only set up a proxy for the search endpoint below.
+  router.post(
+    skipBodyValidation({
+      path: '/api/app_search/search-ui/api/as/v1/engines/{engineName}/search.json',
```

Comment:
Actually sorry, just wanted to leave a comment on this: do we really need to version this or make it look exactly like AS's endpoint? Any thoughts on

```suggestion
      path: '/api/app_search/engines/{engineName}/search-ui/search.json',
```

Instead to better match our other Kibana API endpoints?

Reply:
Sorry, I wondered if my comment above wasn't clear, I think this is evidence that it was not.

Search UI always posts its search query to:
`{whatever base path you provide in configuration}/api/as/v1/engines/{engineName}/search.json`.

I *could* update Search UI and the javascript client to accept a full path as configuration, and not just the base path, but that would be the long fix. I didn't anticipate this issue the last time I updated search-ui.

Reply:
Ohhh gotcha gotcha. No worries in that case, my bad for failing to understand!

Reply:
It's not your bad, it's mine for poor writing 😢